openedx/core/djangoapps/user_api/accounts/tests/test_views.py

```diff
@@ -817,3 +819,69 @@ def test_update_account_settings_rollback(self, mock_email_change):
         data = response.data
         self.assertEqual(old_email, data["email"])
         self.assertEqual(u"m", data["gender"])
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Account APIs are only supported in LMS')
+class TestAccountDeactivation(TestCase):
+    """
+    Tests the account deactivation endpoint.
+    """
+    PASSWORD = 'test'
+
+    def setUp(self):
+        super(TestAccountDeactivation, self).setUp()
+        self.superuser = SuperuserFactory(password=self.PASSWORD)
+        self.staff_user = AdminFactory(password=self.PASSWORD)
+        self.test_user = UserFactory()
+        self.url = reverse('accounts_deactivation', kwargs={'username': self.test_user.username})
+        self.client = APIClient()
+
+    def assert_activation_status(self, client, expected_status=status.HTTP_200_OK, expected_activation_status=False):
+        """
+        Helper function for making a request to the deactivation endpoint, and asserting the status.
+
+        Args:
+            client(APIClient): Client with which the request is made.
+            expected_status(int): Expected request's response status.
+            expected_activation_status(bool): Expected user has_usable_password attribute value.
+        """
+        response = client.post(self.url)
+        self.assertEqual(response.status_code, expected_status)
+        test_user = User.objects.get(username=self.test_user.username)
+        self.assertEqual(test_user.has_usable_password(), expected_activation_status)
+
+    def test_superuser_deactivates_user(self):
+        """
+        Verify a user is deactivated when a superuser posts to the deactivation endpoint.
+        """
+        self.client.login(username=self.superuser.username, password=self.PASSWORD)
+        self.assertTrue(self.test_user.has_usable_password())
+        self.assert_activation_status(self.client)
+
+    def test_user_with_permission_deactivates_user(self):
+        """
+        Verify a user is deactivated when a user with permission posts to the deactivation endpoint.
+        """
+        user = UserFactory()
+        permission = PermissionFactory(
+            codename='can_deactivate_users',
+            content_type=ContentTypeFactory(
+                app_label='student'
+            )
+        )
+        user.user_permissions.add(permission)
+        self.client.login(username=user.username, password=self.PASSWORD)
+        self.assertTrue(self.test_user.has_usable_password())
+        self.assert_activation_status(self.client)
+
+    def test_non_staff_rejection(self):
+        """
+        Verify a non-staff user is rejected from posting to the deactivation endpoint.
```

Comment:
`Verify unauthorized users cannot deactivate accounts.`

Technically, a non-staff user could be granted permission to deactivate accounts. It's not likely; but, if we avoid conflating the staff bit with the ability to deactivate accounts, we can avoid future confusion.